ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+                    if "platform" in platform_item and "speeds" in platform_item:
+                        platform = platform_item
+                    else:
+                        platform = {"platform": platform_item, "speeds": platform_speed_groups[platform_item]}
+                    if platform['platform'] == switch_platform:
+                        speeds = self._convert_dicts(get(platform, 'speeds'), 'speed', 'speed_groups')
+                        for speed in self._natural_sort(speeds, 'speed'):
+                            for speed_group in speed['speed_groups']:
+                                tmp_speed_groups.update({speed_group: speed['speed']})
+            if tmp_speed_groups:
+                hardware = {'speed_groups': {}}
+                for speed_group in self._natural_sort(tmp_speed_groups):
+                    hardware['speed_groups'].update({speed_group: {'serdes': tmp_speed_groups[speed_group]}})
+                return hardware
+        return None
+
+    @cached_property
+    def daemon_terminattr(self):
+        cvp_instance_ip = get(self._hostvars, 'cvp_instance_ip')
+        cvp_instance_ips = get(self._hostvars, 'cvp_instance_ips')
+        if cvp_instance_ip is not None or cvp_instance_ips is not None:
+            cvp_on_prem = {'ips': []}
+            cvaas = {'ips': []}
+            if cvp_instance_ip is not None:
+                if "arista.io" in cvp_instance_ip:
+                    cvaas['ips'].append(cvp_instance_ip)
+                else:
+                    cvp_on_prem['ips'].append(cvp_instance_ip)
+            elif cvp_instance_ips is not None:
+                for cvp_ip in cvp_instance_ips:
+                    if "arista.io" in cvp_ip:
+                        cvaas['ips'].append(cvp_ip)
+                    else:
+                        cvp_on_prem['ips'].append(cvp_ip)
+
+            daemon_terminattr = {'cvaddrs': []}
+            if cvp_on_prem['ips']:
+                for cvp_instance_ip in cvp_on_prem['ips']:
+                    daemon_terminattr['cvaddrs'].append(f"{cvp_instance_ip}:"
+                                                        f"{get(self._hostvars, 'terminattr_ingestgrpcurl_port')}")
+                daemon_terminattr.update({'cvauth': {'method': 'key', 'key': get(self._hostvars,
+                                                                                 'cvp_ingestauth_key')}})
+            elif cvaas['ips']:
+                for cvp_instance_ip in cvaas['ips']:
+                    daemon_terminattr['cvaddrs'].append(f"{cvp_instance_ip}:443")
+                daemon_terminattr.update({'cvauth': {'method': 'token-secure',
+                                                     'token_file': get(self._hostvars, 'cvp_token_file',
+                                                                       '/tmp/cv-onboarding-token')}})
+
+            daemon_terminattr.update({'cvvrf': self._mgmt_interface_vrf})
```

Comment:
Please change to a single update with a larger dict, that you can unfold to still make it more readable.

Reply:
so here I am not sure we are in it for the performance but looking at this: https://discuss.codecademy.com/t/should-we-use-update-or-value-assignment/461981/3 we could also just use regular assignment rather than `update`

Reply:
agreed